tests/tap_tests.rs

```diff
@@ -0,0 +1,149 @@
+/***
+* These tests need to be run as root and currently only work on Linux (and maybe MacOs?)
+*
+* To build and run these tests, run:
+*
+
+ cargo test -F taptest --no-run --test tap_tests |&
```

Comment:
The ampsand at the end should probably be backslash, as an indicator of continueing line.

Reply:
Thanks for the review.  The '|&' is intentional to capture stderr + stdout (this is a /bin/bash'ism), but I agree the backslash would help with readability.  Let me update.

Any other comments on the PR?

Reply:
Today I learnt that `|&` is shorthand for `2>&1 |`.